server/src/main/java/org/opensearch/tasks/TaskResourceTrackingService.java

```diff
@@ -211,7 +212,7 @@ public Map<Long, Task> getResourceAwareTasks() {
         return Collections.unmodifiableMap(resourceAwareTasks);
     }
 
-    private ResourceUsageMetric[] getResourceUsageMetricsForThread(long threadId) {
+    public static ResourceUsageMetric[] getResourceUsageMetricsForThread(long threadId) {
```

Comment:
This class is supposed to be used as singleton object. Making this method static breaks that pattern

Reply:
In that case, let me pass the TaskResourceTrackingService singleton into this class.